File: AMS.API/DTO/Assessor/Assessor.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace AMS.API.DTO.Assessor
{
    public class Assessor
    {
        public Int64 Id { get; set; }

        [Required]
        [StringLength(30)]
        public string IRIS_Id { get; set; }

        [StringLength(20)]
        public string SSC_Id { get; set; }

        [StringLength(20)]
        public string NSDC_Id { get; set; }

        [JsonIgnore]
        public DateTime? CreatedDate { get; set; }

        public Int32 CreatedBy { get; set; }

        [JsonIgnore]
        public DateTime? ModifiedDate { get; set; }

        public Int32 ModifiedBy { get; set; }

        public bool IsActive { get; set; }

        #region All other Assessors related details

        /// <summary>
        /// Assessor basic personal details e.g. FName, DOB, Marital Status etc
        /// </summary>
        public AssessorPersonalDetail PerosnalDetail { get; set; }

        /// <summary>
        /// Assessor basic personal details e.g. FName, DOB, Marital Status etc
        /// </summary>
        public AssessorCommunicationDetail CommunicationDetail { get; set; }

        /// <summary>
        /// Assessor all other details like OdigoStatus, BankName, Number etc
        /// </summary>
        public AssessorOtherDetail OtherDetail { get; set; }

        /// <summary>
        /// Assessor basic unique identity proofs e.g. Aadhar, DL, VoterId etc
        /// </summary>
        public List<AssessorIdProofDetail> IdProofs { get; set; }

        /// <summary>
        /// Assessor professional preferred Job-Roles list
        /// </summary>
        public List<AssessorJobRole> JobRoles { get; set; }

        /// <summary>
        /// Assessor preferred Job-Locations(City) list
        /// </summary>
        public List<AssessorPreferredLocation> PreferredLocations { get; set; }

        #endregion
    }
}

File: AMS.Web/src/app/models/assessor-demographic/assessor.ts
import { AssessorBase } from "./assessor-base";
import { AssessorPersonalDetail } from "./assessor-personal-detail";
import { AssessorCommunicationDetail } from "./assessor-communication-detail";
import { AssessorOtherDetail } from "./assessor-other-detail";
import { AssessorIdProofDetail } from "./assessor-id-proof-detail";
import { AssessorJobRole } from "./assessor-job-role";
import { AssessorPreferredLocation } from "./assessor-preferred-location";

export class Assessor extends AssessorBase {
    constructor(){        
        super();
        
        this.PerosnalDetail = new AssessorPersonalDetail();
        this.CommunicationDetail = new AssessorCommunicationDetail();
        this.OtherDetail = new AssessorOtherDetail();
        this.IdProofs = Array<AssessorIdProofDetail>();
        this.JobRoles = new Array<AssessorJobRole>();
        this.PreferredLocations = new Array<AssessorPreferredLocation>();
    };

    IRIS_Id: string = null;
    SSC_Id: string = null;
    NSDC_Id: string = null;

    //#region All other Assessors related details

    /**
     * Assessor basic personal details e.g. FName, DOB, Marital Status etc
     */
    PerosnalDetail: AssessorPersonalDetail;

    /**
     * Assessor basic personal details e.g. FName, DOB, Marital Status etc
     */
    CommunicationDetail: AssessorCommunicationDetail;

    /**
     * Assessor all other details like OdigoStatus, BankName, Number etc
     */
    OtherDetail: AssessorOtherDetail;

    /**
     * Assessor basic unique identity proofs e.g. Aadhar, DL, VoterId etc
     */
    IdProofs: Array<AssessorIdProofDetail>;

    /**
     * Assessor professional preferred Job-Roles list
     */
    JobRoles: Array<AssessorJobRole>;

    /**
     * Assessor preferred Job-Locations(City) list
     */
    PreferredLocations: Array<AssessorPreferredLocation>;

    //#endregion
}
